### PR TITLE
refactor: refactor wizard bottom bar

### LINKED ISSUE #176: refactor: wizard bottom bar
Until now the bottom bar displayed in the wizard's widgets (accessed either in the onboarding or from the settings menu) was a simple row in a column, instead of a real bottom navigation bar.
The goal of this issue is to refactor the code to use a bottom navigation bar in the wizard.

### PR DESCRIPTION
**Pull request by LeanderCretegny**
Handle the refactor of the bottom bar in the wizard. The bottom bar is now handle in a scaffold as it should be.

To test: run the ci locally and run the code on a android phone, and either go through the onboarding or go in the settings to see the changes made.

This PR closes #176

### CODE REVIEW

**Review by quartztz — CHANGES_REQUESTED** [review]
Thank you for the much needed rework ! I left a few comments but they're all pretty benign and fast to fix. Let me know if you have any questions !

File: lib/views/interests_wizard_view.dart [hunk 1]
```diff
@@ -35,101 +35,130 @@ class _InterestWizardViewState extends State<InterestWizardView> {
     super.dispose();
   }
 
+  void setListState<T>(T item, List<T> list) {
```
[the remark this answers is not in the document]
Comment (LeanderCretegny) [reply lib/views/interests_wizard_view.dart:38]: You are right, but I feel like the name should reflect the use of set state in priority. I would suggest renaming it to `updateListStateOnSelection`.

File: lib/widgets/wizard_widgets.dart [hunk 2]
```diff
@@ -45,37 +38,27 @@ class _WizardSelector extends State<WizardSelector> {
...
-      onCancel: widget.onCancel,
-      showRight: true,
-      rText: widget.buttonText,
-      rOnPressed: () {
-        widget.onPressed(_selectedItems);
-      },
-    );
+            padding:
+                const EdgeInsets.fromLTRB(0.0, UNIT_PADDING * 1.5, 0.0, 0.0),
+            child: FadingEdgeScrollView.fromSingleChildScrollView(
```
[the remark this answers is not in the document]
Comment (LeanderCretegny) [reply lib/widgets/wizard_widgets.dart:43]: For the fading on the top I'm not sure if this package allow it. I will look a bit more into it but can't guarantee that I'll find something.

File: lib/views/alarm_wizard.dart [hunk 3]
```diff
@@ -47,40 +47,41 @@ class _AlarmWizardView extends State<AlarmWizardView> {
...
-            }
-            Navigator.popUntil(context, (route) => !route.hasActiveRouteBelow);
-          },
-        )
       ],
     );
 
+    Widget bottomBar = WizardNavigationBottomBar(
+      showCancel: true,
+      cancelText: "Skip",
```
Comment (Thechi2000) [inline lib/views/alarm_wizard.dart:55]: This needs to be localized (as per #174).

File: lib/views/interests_wizard_view.dart [hunk 4]
```diff
@@ -35,101 +35,130 @@ class _InterestWizardViewState extends State<InterestWizardView> {
...
-      },
       key: const Key("countries-selector"),
     );
 
+    Widget countriesBottomBar = WizardNavigationBottomBar(
+        showCancel: !auth.isOnboardingRequired,
+        onCancel: () {
+          Navigator.pop(context);
+        },
+        rText: "Next",
```
Comment (Thechi2000) [inline lib/views/interests_wizard_view.dart:69]: This needs to be localized (as per #174).

File: lib/views/interests_wizard_view.dart [hunk 5]
```diff
@@ -35,101 +35,130 @@ class _InterestWizardViewState extends State<InterestWizardView> {
...
     );
 
+    Widget interestsBottomBar = WizardNavigationBottomBar(
+        showCancel: !auth.isOnboardingRequired,
+        onCancel: () {
+          setState(() {
+            _step = WizardStep.CITIES;
+          });
+        },
+        rText: auth.isOnboardingRequired ? "Next" : "Finish",
```
Comment (Thechi2000) [inline lib/views/interests_wizard_view.dart:117]: This needs to be localized (as per #174).

File: lib/views/providers_wizard_view.dart [hunk 6]
```diff
@@ -62,27 +62,29 @@ class _ProvidersWizardView extends State<ProvidersWizardView> {
...
-            }
-          },
-        ),
       ]);
     }
 
-    return WizardScaffold(body: Material(child: body));
+    Widget bottomBar = WizardNavigationBottomBar(
+      showCancel: !auth.isOnboardingRequired,
+      rText: auth.isOnboardingRequired ? "Next" : "Done",
```
Comment (Thechi2000) [inline lib/views/providers_wizard_view.dart:70]: This needs to be localized (as per #174).

File: lib/views/alarm_wizard.dart [hunk 7]
```diff
@@ -47,40 +47,41 @@ class _AlarmWizardView extends State<AlarmWizardView> {
...
+      onCancel: () async {
+        if (auth.isOnboardingRequired) {
+          await auth.setOnboardingIsDone();
+        }
+        if (context.mounted) {
+          Navigator.popUntil(context, (route) => !route.hasActiveRouteBelow);
+        }
+      },
+      showRight: true,
+      rText: "Validate",
```
Comment (Thechi2000) [inline lib/views/alarm_wizard.dart:65]: This needs to be localized (as per #174).

File: lib/views/interests_wizard_view.dart [hunk 8]
```diff
@@ -35,101 +35,130 @@ class _InterestWizardViewState extends State<InterestWizardView> {
...
     );
 
+    Widget citiesBottomBar = WizardNavigationBottomBar(
+        showCancel: !auth.isOnboardingRequired,
+        onCancel: () {
+          setState(() {
+            _step = WizardStep.COUNTRIES;
+          });
+        },
+        rText: "Next",
```
Comment (Thechi2000) [inline lib/views/interests_wizard_view.dart:93]: This needs to be localized (as per #174).

**Review by Thechi2000 — COMMENTED** [review]
I like the new UI, thanks ! Just a few comments about localization, now that it has been merged.

Also, what do you think about adding a `Back` button to go to the previous selector (e.g. from interests to cities) ? Could that be implemented easily ? Don't bother if it is too complex, it's just a suggestion since you are working on refactoring this.

**Comment by LeanderCretegny** [comment]
Regarding the back button, it is present (but I am removing it in the #188). User can still go back by swiping to the left, but I don't think it is a behavior that should be encouraged.

**Review by Thechi2000 — APPROVED** [review]
LGTM !

**Review by quartztz — APPROVED** [review]
All good to me as well ! can be merged

**Pull request merged**